server/lib/src/valueset/eckey.rs

```diff
@@ -0,0 +1,207 @@
+use std::iter::{self};
+
+use crate::be::dbvalue::DbValueSetV2;
+use crate::prelude::ValueSetT;
+use crate::repl::proto::ReplAttrV1;
+use crate::value::{PartialValue, PubOrPrivEcKey, SyntaxType, Value};
+use kanidm_proto::v1::OperationError;
+use openssl::ec::EcKey;
+use openssl::pkey::{Private, Public};
+use openssl::sha::sha256;
+
+use super::ValueSet;
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct ValueSetEcKey {
+    key: PubOrPrivEcKey,
```

Comment:
I think given the above, this could be change to rehydrate the vaue from the db/repl as a private key, but then the struct should have

```
priv: EcKey<Private>
public: EcKey<Public>
```
This way we only have to do the private -> public extract once on load from disk or update, and then we can rely on the entries being Arc counted and we can return references etc.